lib/spack/spack/environment/environment.py

```diff
@@ -1527,33 +1527,14 @@ def _concretize_separately(self, tests=False):
 
         # Add specs in original order
         batch.sort(key=lambda x: x[0])
-        by_hash = {}  # for attaching information on test dependencies
+        unified_specs = spack.solver.asp.ConcreteSpecsByHash()
```

Comment:
I don't think this is correct cause the environment already has old concrete specs? Also the term unify is overloaded. Maybe deduplicate (by dag hash)?

Reply:
> I don't think this is correct cause the environment already has old concrete specs?

Fair point. Makes me wonder if we should use this class as an internal member of the environment.